src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -2425,10 +2425,8 @@ ClientChannel::LoadBalancedCall::LbCallState::GetCallAttribute(
     UniqueTypeName type) {
   auto* service_config_call_data = static_cast<ServiceConfigCallData*>(
       lb_call_->call_context()[GRPC_CONTEXT_SERVICE_CONFIG_CALL_DATA].value);
-  auto& call_attributes = service_config_call_data->call_attributes();
-  auto it = call_attributes.find(type);
-  if (it == call_attributes.end()) return absl::string_view();
-  return it->second;
+  return ServiceConfigCallData::StringViewAttribute::FromCallData(
```

Comment:
Since the whole point of call attributes is to provide a mechanism for communicating arbitrary information to the LB policies, and the point of this PR is to allow call attributes to be types other than string, I think we need to change the `ClientChannel::LoadBalancedCall::LbCallState::GetCallAttribute()` method to return `CallAttributeInterface*` instead of `absl::string_view`.  It needs to be up to the individual LB policy to decide what type to down-cast to.

Note that this means that it won't be possible to use `StringViewAttribute::FromCallData()` in LB policies, since they will need to use `ClientChannel::LoadBalancedCall::LbCallState::GetCallAttribute()` instead of `GetCallAttribute::GetCallAttribute()`.  Given that, see my comment elsewhere about whether `StringViewAttribute` is the best way to go here.

Reply:
Done